src/nodes.coffee

```diff
@@ -3255,7 +3255,15 @@ exports.Throw = class Throw extends Base
   makeReturn: THIS
 
   compileNode: (o) ->
-    fragments = @expression.compileToFragments o
```

Comment:
Is there any chance that changing this to `@expression.compileToFragments o, LEVEL_LIST` would work as well?

Reply:
Looks like the tests still pass with:

```diff
diff --git a/src/nodes.coffee b/src/nodes.coffee
index 677d4df0..f3b89902 100644
--- a/src/nodes.coffee
+++ b/src/nodes.coffee
@@ -3255,7 +3255,7 @@ exports.Throw = class Throw extends Base
   makeReturn: THIS
 
   compileNode: (o) ->
-    fragments = @expression.compileToFragments o
+    fragments = @expression.compileToFragments o, LEVEL_LIST
     unshiftAfterComments fragments, @makeCode 'throw '
     fragments.unshift @makeCode @tab
     fragments.push @makeCode ';'
```

Reply:
That would also fiw `throw try ...`.

Reply:
Er, fix*.

Reply:
`@expression.compileToFragments o, LEVEL_LIST` is interesting. Care to explain what that means? I've never fully understood the second parameter there.

Reply:
Since CoffeeScript generates a JS string directly and not a JS ast, we carry around a notion of what context the node is gonna be compiled in, so we can generate expr/statement, surround with parentheses, etc not to mess with the precedence/... (like compiling `{a:5}` toplevel)

Reply:
That's a good way of putting it, re: there being no JS AST. It's exactly designed as a way of dealing with the situation in this PR (and #4644) - namely that a given JS expression (e.g. `If`, `Obj`, `Op` etc.) that's a child of another expression needs to be compiled in a certain way in order to be valid JS (e.g. wrapped in parens, closures, etc.).